### ISSUE TITLE
Rendering may fail on non-corrupt chunks

### ISSUE THREAD

**Issue by piegamesde** [overviewer/Minecraft-Overviewer#1639]
Look at <https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/world.py#L1137>. If `bits_per_value > 12`, this will throw a `CorruptChunkError`.

But the maximum bits per value that may be in a tag actually is 13. This is because air always gets saved at palette index 0, no matter what, for some reasons. If you now have a section with completely different blocks, and none of them is air, you will end with a palette containing 4097 items, and thus need 13 bit to address it.

**Comment by CounterPillow** [comment]
Do you have an example file generated by Minecraft that uses a 13 bit palette?

**Comment by piegamesde** [comment]
@CounterPillow This would require putting 4 thousand distinct blocks into a single section, and I gave up just after one hundred\*.

But I did look at the source code, and as usual, Minecraft does a few funky things. The relevant code is in `net.minecraft.world.chunk.BlockStateContainer#readChunkPalette(ListNBT, long[])` and `net.minecraft.util.BitArray`.

The `BitArray` constructor has a range check, but it was not what I excpected:

```java
   public BitArray(int p_i47901_1_, int p_i47901_2_, long[] p_i47901_3_) {
      Validate.inclusiveBetween(1L, 32L, (long)p_i47901_1_);
      […]
   }
```

When calling that constructor in `readChunkPalette`, that parameter either is `Math.max(4, MathHelper.log2DeBruijn(p_222642_1_.size()))` or `p_222642_2_.length * 64 / 4096`. So apparently, there are no other range checks than that `1..32`.

-------

\* I just now came to the idea to fill a section with stone.

![image](https://user-images.githubusercontent.com/14054505/63850840-4f582e80-c995-11e9-9d14-a6650732d247.png)

Minecraft has generated 2 palette entries, one for stone, and one for air, even if there is no air inside. So a section with 4067 should be possible.

-------

**TL;DR**:

- Minecraft apparently supports palette sizes from 1 to 32 bits
- But it will never use less than 4 bits (checked through `Math.max(4, …)`)
- A valid section may have a 13 bit palette
- And I don't know how one would create a section with even more bits, maybe Minecraft could use this for some optimizations or for coming features?
